WordPress/Classes/ViewRelated/Menus/Views/MenuItemAbstractPostsViewController.m

```diff
@@ -0,0 +1,201 @@
+                     success:^(NSArray *posts) {
+                         [self didFinishSyncingPosts:posts options:options];
+                     } failure:^(NSError *error) {
+                         [self didFinishSyncingPosts:nil options:options];
+                         [self showLoadingErrorMessageForResults];
+                     }];
+}
+
+#pragma mark - TableView methods
+
+- (void)configureSourceCell:(MenuItemSourceCell *)cell forIndexPath:(NSIndexPath *)indexPath
+{
+    AbstractPost *post = [self.resultsController objectAtIndexPath:indexPath];
+    if ([self itemTypeMatchesSourceItemType] && post.postID.integerValue == [self.item.contentID integerValue]) {
+        cell.sourceSelected = YES;
+    } else {
+        cell.sourceSelected = NO;
+    }
+    
+    NSString *postTitle = post.titleForDisplay;
+    if (!postTitle.length) {
+        postTitle = NSLocalizedString(@"(Untitled Post)", @"Menus title label text for a post that has no set title.");
+    }
+    [cell setTitle:postTitle];
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [super tableView:tableView didSelectRowAtIndexPath:indexPath];
+    
+    AbstractPost *post = [self.resultsController objectAtIndexPath:indexPath];
+    [self setItemSourceWithContentID:post.postID name:post.titleForDisplay];
+    
+    [self deselectVisibleSourceCellsIfNeeded];
+    
+    MenuItemSourceCell *selectedCell = [tableView cellForRowAtIndexPath:indexPath];
+    selectedCell.sourceSelected = YES;
+}
+
+#pragma mark - MenuItemSourcePostAbstractViewSubclass
+
+- (Class)entityClass
+{
+    AssertSubclassMethod();
+    return nil;
+}
+
+- (PostServiceSyncOptions *)syncOptions
+{
+    PostServiceSyncOptions *options = [[PostServiceSyncOptions alloc] init];
+    options.statuses = @[PostStatusPublish, PostStatusPrivate];
+    options.number = @(PostServiceDefaultNumberToSync);
+    return options;
+}
+
+- (void)didFinishSyncingPosts:(NSArray *)posts options:(PostServiceSyncOptions *)options
+{
+    if (posts) {
+        self.numberOfSyncedPosts = self.numberOfSyncedPosts + posts.count;
+        self.additionalPostsAvailableForSync = posts.count >= options.number.unsignedIntegerValue;
+        self.resultsController.fetchRequest.fetchLimit = self.numberOfSyncedPosts;
+        [self performResultsControllerFetchRequest];
+        [self.tableView reloadData];
+    }
+    [self hideLoadingSourcesIndicator];
+}
+
+#pragma mark - paging
+
+- (void)scrollingWillDisplayEndOfTableView:(UITableView *)tableView
+{
+    if (!self.additionalPostsAvailableForSync || [self searchBarInputIsActive]) {
+        return;
+    }
+    [self showLoadingSourcesIndicator];
+    PostService *service = [[PostService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    PostServiceSyncOptions *options = [self syncOptions];
+    options.offset = @(self.resultsController.fetchedObjects.count);
+    [service syncPostsOfType:[self sourceItemType]
```

Comment:
If a user is on a slow connection its possible for them to trigger multiple syncs by scrolling down, then back up, then back down, then back up then back down, etc. 
Maybe check for an in-progress sync before starting a new one.